### PR TITLE
add set time zone sometimezone as a exception while parsing keyword::set

### LINKED ISSUE #303: can not parse "SET TIME ZONE 'UTC'"
It was normal to parse "SET TIME ZONE 'UTC'" a few days ago. Today I tried to parse "SET TIME ZONE 'UTC'" and there was an error.
```
pub fn sql_parse_test() {
    use sqlparser::dialect::GenericDialect;
    use sqlparser::parser::Parser;

    let sql = r#"set time zone 'UTC'"#;

    let dialect = GenericDialect {};

    let ast = Parser::parse_sql(&dialect, sql).unwrap();

    println!("AST: {:?}", ast);
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_sql_parse() {
        sql_parse_test();
    }
}
```
```

called `Result::unwrap()` on an `Err` value: ParserError("Expected equals sign or TO, found: zone")
thread 'func::sql_parse_test::tests::test_sql_parse' panicked at 'called `Result::unwrap()` on an `Err` value: ParserError("Expected equals sign or TO, found: zone")', src\func\sql_parse_test.rs:9:48
```

### PR DESCRIPTION
**Pull request by waitingkuo**
closes #303

### CODE REVIEW

File: src/parser.rs [hunk 1]
```diff
@@ -4553,7 +4553,10 @@ impl<'a> Parser<'a> {
                 charset_name,
                 collation_name,
             })
-        } else if self.consume_token(&Token::Eq) || self.parse_keyword(Keyword::TO) {
+        } else if (self.consume_token(&Token::Eq) || self.parse_keyword(Keyword::TO))
+            || (variable == ObjectName(vec!["TIMEZONE".into()]))
+        {
+            // when the object name is TIMEZONE, we support `SET TIMEZONE 'UTC'` without Eq sign or TO
```
Comment (waitingkuo) [inline src/parser.rs:4559]: @alamb @435201823

`SET TIMEZONE 'UTC'` is an exceptional case in posetgresql 
not sure whether it's a good idea to add it here

**Comment by waitingkuo** [comment]
i'd like to make sure this implement make sense first, then i'll add some test cases later

File: src/parser.rs (hunk 1 above)
Comment (waitingkuo) [inline src/parser.rs:4560]: @alamb @435201823

SET TIMEZONE 'UTC' is an exceptional case in posetgresql
not sure whether it's a good idea to add it here

File: src/parser.rs [hunk 2]
```diff
@@ -4553,7 +4553,11 @@ impl<'a> Parser<'a> {
                 charset_name,
                 collation_name,
             })
-        } else if self.consume_token(&Token::Eq) || self.parse_keyword(Keyword::TO) {
+        } else if self.consume_token(&Token::Eq)
```
Comment (AugustoFKL) [inline src/parser.rs:4556]: Having tests would make it easier, but, tbh, I guess the SET TIME ZONE could be an individual command, no?

It is a singular command in many dialects. Having this manual conversion seems weird to me, considering that we can't have `SET TIME ZONE = ...` (I'm basing myself on [PostgreSQLs Docs](https://www.postgresql.org/docs/current/sql-set.html))

File: src/ast/mod.rs [hunk 3]
```diff
@@ -1271,6 +1271,11 @@ pub enum Statement {
         variable: ObjectName,
         value: Vec<Expr>,
     },
+    /// SET TIME ZONE <value>
+    ///
+    /// Note: this is a PostgreSQL-specific statements
+    /// SET TIME ZONE <value> is an alias for SET timezone TO <value> in PostgreSQL
+    SetTimeZone { local: bool, value: Expr },
```
Comment (waitingkuo) [inline src/ast/mod.rs:1278]: add a new Statement SetTimeZone to reflect the difference between `SET TIME ZONE TO 'UTC'` and `SET TIME ZONE 'UTC'`

File: tests/sqlparser_common.rs [hunk 4]
```diff
@@ -4979,6 +4979,19 @@ fn parse_set_time_zone() {
...
+    println!("wew");
+    match verified_stmt("SET TIME ZONE 'UTC'") {
+        Statement::SetTimeZone { local, value } => {
+            assert!(!local);
+            println!("wew");
+            assert_eq!(value, Expr::Value(Value::SingleQuotedString("UTC".into())));
+        }
+        _ => unreachable!(),
+    }
+}
```
Comment (waitingkuo) [inline tests/sqlparser_common.rs:4993]: add a test case for `set time zone 'utc'`

File: src/parser.rs [hunk 5]
```diff
@@ -4573,6 +4573,15 @@ impl<'a> Parser<'a> {
...
             }
+        } else if variable.to_string().eq_ignore_ascii_case("TIMEZONE") {
+            // for some db (e.g. postgresql), SET TIME ZONE <value> is an alias for SET TIMEZONE [TO|=] <value>
+            match self.parse_expr() {
+                Ok(expr) => Ok(Statement::SetTimeZone {
+                    local: modifier == Some(Keyword::LOCAL),
+                    value: expr,
+                }),
+                _ => self.expected("timezone value", self.peek_token())?,
+            }
```
Comment (waitingkuo) [inline src/parser.rs:4584]: this captures the `SET TIME ZONE <value>` case

**Review by waitingkuo — COMMENTED** [review]
@AugustoFKL @alamb @step-baby 
thank you for the comments and feedbacks

here's the update:

1. add a new statement `Statement::SetTimeZone`
2. `SET <variable> [TO|=] <value>` is parsed as `Statement::SetVariable`
4. `SET TIME ZONE [TO|=] <value>` is parsed as `Statement::SetVariable`
5. `SET TIME ZONE <value>` is parsed as `Statement::SetTimeZone`
6. `SET <non-time-zone-variable> <value>` doesn't work

File: tests/sqlparser_common.rs [hunk 6]
```diff
@@ -4979,6 +4979,19 @@ fn parse_set_time_zone() {
     one_statement_parses_to("SET TIME ZONE TO 'UTC'", "SET TIMEZONE = 'UTC'");
 }
 
+#[test]
+fn parse_set_time_zone_alias() {
+    println!("wew");
+    match verified_stmt("SET TIME ZONE 'UTC'") {
+        Statement::SetTimeZone { local, value } => {
+            assert!(!local);
+            println!("wew");
```
Comment (alamb) [inline tests/sqlparser_common.rs:4988]: is the `println` meant to be left in?

**Review by alamb — APPROVED** [review]
Other than the `println` looks good to me -- thank you @waitingkuo

File: tests/sqlparser_common.rs (hunk 6 above)
Comment (waitingkuo) [reply tests/sqlparser_common.rs]: updated, thank you @alamb

**Pull request merged**